### PR TITLE
add paddle ppo, add ppo example on Mujoco env

### PR DESCRIPTION
**Pull request by swag1ong**

### CODE REVIEW

File: benchmark/torch/ppo/train.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (rical730) [inline benchmark/torch/ppo/train.py:1]: Good, please replace all the 2018 with 2021.

File: benchmark/torch/ppo/train.py [hunk 2]
```diff
@@ -14,115 +14,147 @@
...
+            summary.add_scalar('ppo/mean_validation_rewards', eval_mean_reward,
+                               (j + 1) * NUM_STEPS)
 
 
 if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='RL')
+    parser.add_argument(
+        '--seed', type=int, default=1, help='random seed (default: 1)')
+    parser.add_argument(
+        '--batch_size',
```
Comment (rical730) [inline benchmark/torch/ppo/train.py]: Please remove this argument, and declare it as a global variable.

File: examples/PPO/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+### Benchmark result
+
+<p align="center">
+<img src=".benchmark/ppo.png" alt="result"/>
+</p>
+
+## How to use
+### Dependencies:
++ [paddle>=2.0.0](https://github.com/PaddlePaddle/Paddle)
++ [parl](https://github.com/PaddlePaddle/PARL)
```
Comment (rical730) [inline examples/PPO/README.md]: parl>=2.0.2

File: examples/PPO/train.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    while len(eval_episode_rewards) < 10:
+        action = agent.predict(obs)
+
+        # Observe reward and next obs
+        obs, _, done, info = eval_env.step(action)
+
+        if 'episode' in info.keys():
+            eval_episode_rewards.append(info['episode']['r'])
+
+    eval_env.close()
```
Comment (rical730) [inline examples/PPO/train.py:61]: Can we remove this line ?

File: examples/PPO/train.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            summary.add_scalar('ppo/mean_validation_rewards', eval_mean_reward,
+                               (j + 1) * NUM_STEPS)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='RL')
+    parser.add_argument(
+        '--seed', type=int, default=616, help='random seed (default: 1)')
+    parser.add_argument(
+        '--batch_size',
```
Comment (rical730) [inline examples/PPO/train.py]: Please remove this argument, and declare it as a global variable.

File: examples/PPO/train.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            ob_rms = get_vec_normalize(env).ob_rms
+            eval_mean_reward = evaluate(agent, ob_rms, args.env, args.seed)
+            summary.add_scalar('ppo/mean_validation_rewards', eval_mean_reward,
+                               (j + 1) * NUM_STEPS)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='RL')
+    parser.add_argument(
+        '--seed', type=int, default=616, help='random seed (default: 1)')
```
Comment (rical730) [inline examples/PPO/train.py]: help='random seed (default: 616)'

File: benchmark/torch/ppo/train.py [hunk 7]
```diff
@@ -14,115 +14,147 @@
...
+    while len(eval_episode_rewards) < 10:
+        action = agent.predict(obs)
+
+        # Observe reward and next obs
+        obs, _, done, info = eval_env.step(action)
 
+        if 'episode' in info.keys():
+            eval_episode_rewards.append(info['episode']['r'])
+
+    eval_env.close()
```
Comment (rical730) [inline benchmark/torch/ppo/train.py:61]: Can we remove this line ?

File: parl/env/mujoco_wrappers.py [hunk 8]
```diff
@@ -163,17 +155,68 @@ def train(self):
...
-def make_env(env_name, seed, gamma):
+    Args:
+        env_name (str): Mujoco env name
+        seed (int): random seed
+        gamma (float or None): discounting factor, if test then gamma = None
+        test (bool): True if test else False
+        ob_rms (None or np.array): ob_rms from training environment, not None only when test is True
+    """
     env = gym.make(env_name)
     env.seed(seed)
```
Comment (rical730) [inline parl/env/mujoco_wrappers.py]: please make env and set seed in train.py

File: parl/env/mujoco_wrappers.py [hunk 9]
```diff
@@ -21,6 +22,9 @@
 
 
 class TimeLimitMask(gym.Wrapper):
```
Comment (rical730) [inline parl/env/mujoco_wrappers.py]: TimeLimitMask-->TimeLimitMaskEnv

File: parl/env/mujoco_wrappers.py [hunk 10]
```diff
@@ -93,22 +91,10 @@ def update_from_moments(self, batch_mean, batch_var, batch_count):
...
-    m_a = var * count
-    m_b = batch_var * batch_count
-    M2 = m_a + m_b + np.square(delta) * count * batch_count / tot_count
-    new_var = M2 / tot_count
-    new_count = tot_count
-
-    return new_mean, new_var, new_count
-
-
 class VecNormalize(gym.Wrapper):
```
Comment (rical730) [inline parl/env/mujoco_wrappers.py]: VecNormalize --> VecNormalizeEnv

File: parl/env/mujoco_wrappers.py [hunk 11]
```diff
@@ -163,17 +155,68 @@ def train(self):
...
+    m_a = var * count
+    m_b = batch_var * batch_count
+    M2 = m_a + m_b + np.square(delta) * count * batch_count / tot_count
+    new_var = M2 / tot_count
+    new_count = tot_count
+
+    return new_mean, new_var, new_count
+
 
+def make_env(env_name, seed, gamma, test=False, ob_rms=None):
```
Comment (rical730) [inline parl/env/mujoco_wrappers.py]: make_env --> wrap_rms(env, gamma, test=False, obs_rms=None)

File: parl/env/mujoco_wrappers.py [hunk 12]
```diff
@@ -163,17 +155,68 @@ def train(self):
...
+        self.training = False
+
+
+def get_vec_normalize(venv):
+    """ Fetch VecNormalize class
+
+    Args:
+        venv (gym.Wrapper): current env
+    """
+    if isinstance(venv, VecNormalize):
```
Comment (rical730) [inline parl/env/mujoco_wrappers.py]: VecNormalize --> VecNormalizeEnv

File: examples/PPO/train.py [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+            print(
+                "Updates {}, num timesteps {},\n Last {} training episodes: mean/median reward {:.1f}/{:.1f}, min/max reward {:.1f}/{:.1f}\n"
+                .format(j, total_num_steps, len(episode_rewards),
+                        np.mean(episode_rewards), np.median(episode_rewards),
+                        np.min(episode_rewards), np.max(episode_rewards),
+                        dist_entropy, value_loss, action_loss))
+
+        if (args.test_every_steps is not None and len(episode_rewards) > 1
+                and j % args.test_every_steps == 0):
+            ob_rms = get_vec_normalize(env).ob_rms
```
Comment (rical730) [inline examples/PPO/train.py]: Can we get obs_rms more appropriately？

**Review by rical730 — APPROVED** [review]
LGTM~
